mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,330 @@
+     [P0] Summary plot
+"""
+
+from PIL.Image import Image, open as open_image
+
+
+class ImageEvaluationArtifact(EvaluationArtifact):
+    def save(self, output_artifact_path):
+        self._content.save(output_artifact_path)
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = open_image(local_artifact_path)
+        return self._content
+
+
+class CsvEvaluationArtifact(EvaluationArtifact):
+    def save(self, output_artifact_path):
+        self._content.to_csv(output_artifact_path, index=False)
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = pd.read_csv(local_artifact_path)
+        return self._content
+
+
+_DEFAULT_SAMPLE_ROWS_FOR_SHAP = 2000
+
+
+class DefaultEvaluator(ModelEvaluator):
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    def _log_metrics(self, run_id, metrics, dataset_name):
+        """
+        Helper method to log metrics into specified run.
+        """
+        client = mlflow.tracking.MlflowClient()
+        timestamp = int(time.time() * 1000)
+        client.log_batch(
+            run_id,
+            metrics=[
+                Metric(key=f"{key}_on_{dataset_name}", value=value, timestamp=timestamp, step=0)
+                for key, value in metrics.items()
+            ],
+        )
+
+    def _log_image_artifact(
+        self, artifacts, temp_dir, do_plot, run_id, artifact_name, dataset_name
+    ):
+        client = mlflow.tracking.MlflowClient()
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.png"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
+        client.log_artifact(run_id, artifact_file_local_path)
+        artifact = ImageEvaluationArtifact(uri=get_artifact_uri(run_id, artifact_file_name))
+        artifact.load(artifact_file_local_path)
+        artifacts[artifact_file_name] = artifact
+
+    def _log_pandas_df_artifact(
+        self, artifacts, temp_dir, pandas_df, run_id, artifact_name, dataset_name
+    ):
+        client = mlflow.tracking.MlflowClient()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.csv"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pandas_df.to_csv(artifact_file_local_path, index=False)
+        client.log_artifact(run_id, artifact_file_local_path)
+        artifact = CsvEvaluationArtifact(
+            uri=get_artifact_uri(run_id, artifact_file_name),
+            content=pandas_df,
+        )
+        artifact.load(artifact_file_local_path)
+        artifacts[artifact_file_name] = artifact
+
+    def _log_model_explainality(
+            self, artifacts, temp_dir, model, X, dataset_name, feature_names, run_id, evaluator_config
+    ):
+        sample_rows = evaluator_config.get('explainality.nsamples', _DEFAULT_SAMPLE_ROWS_FOR_SHAP)
```

Comment:
I prefer to set default sample rows instead of default sample ratio, because, the dataset size varies, if we set the fixed sample rows by default, the default shap calculation time is controllable.

Reply:
```suggestion
        sample_rows = evaluator_config.get('explainability.nsamples', _DEFAULT_SAMPLE_ROWS_FOR_SHAP)
```

Reply:
A fixed number of sample rows is definitely a good idea here instead of % of total rows with a floor value (as you mentioned, with a VERY large data set this can be problematic) or trying to estimate in-memory size of an object (a very complex task to get marginally correct). 
Good idea!